### PR TITLE
feat: add nunjucks grammar

### PR DESCRIPTION
**Pull request by LM1207**
- [x] Add a test if possible
- [x] Format all commit messages with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I have read docs for [adding a language](https://github.com/shikijs/shiki/blob/main/docs/languages.md#adding-grammar).
- [x] I have searched around and this is the most up-to-date, actively maintained version of the language grammar.
- [x] I have added a sample file that includes a variety of language syntaxes and succinctly captures the idiosyncrasy of a language. See [docs](https://github.com/shikijs/shiki/blob/main/docs/languages.md#adding-grammar) for requirement.

### CODE REVIEW

**Comment by octref** [comment]
Hi – you can't really edit the `languages.ts` manually. Changes will be overwritten.

File: packages/shiki/src/languages.ts [hunk 1]
```diff
@@ -821,7 +821,7 @@ export const languages: ILanguageRegistration[] = [
     path: 'markdown.tmLanguage.json',
     displayName: 'Markdown',
     aliases: ['md'],
-    embeddedLangs: ['css', 'html', 'ini', 'java', 'lua', 'make', 'perl', 'r', 'ruby', 'php', 'sql', 'vb', 'xml', 'xsl', 'yaml', 'bat', 'clojure', 'coffee', 'c', 'cpp', 'diff', 'docker', 'git-commit', 'git-rebase', 'go', 'groovy', 'pug', 'javascript', 'json', 'jsonc', 'less', 'nunjucks', 'objective-c', 'swift', 'scss', 'raku', 'powershell', 'python', 'julia', 'rust', 'scala', 'shellscript', 'typescript', 'tsx', 'csharp', 'fsharp', 'dart', 'handlebars', 'erlang', 'elixir', 'latex', 'bibtex']
+    embeddedLangs: ['css', 'html', 'ini', 'java', 'lua', 'make', 'perl', 'r', 'ruby', 'php', 'sql', 'vb', 'xml', 'xsl', 'yaml', 'bat', 'clojure', 'coffee', 'c', 'cpp', 'diff', 'docker', 'git-commit', 'git-rebase', 'go', 'groovy', 'pug', 'javascript', 'json', 'jsonc', 'less', 'objective-c', 'swift', 'scss', 'raku', 'powershell', 'python', 'julia', 'rust', 'scala', 'shellscript', 'typescript', 'tsx', 'csharp', 'fsharp', 'dart', 'handlebars', 'erlang', 'elixir', 'latex', 'bibtex']
```
Comment (octref) [inline packages/shiki/src/languages.ts]: this is dictated by markdown grammar, not nunjucks.

File: packages/shiki/src/languages.ts [hunk 2]
```diff
@@ -893,7 +893,7 @@ export const languages: ILanguageRegistration[] = [
     displayName: 'Nunjucks',
     samplePath: 'nunjucks.sample',
     aliases: ['njk'],
-    embeddedLangs: ['yaml', 'javascript', 'css', 'html']
+    embeddedLangs: ['yaml']
```
Comment (octref) [inline packages/shiki/src/languages.ts:896]: this is dependent on the grammar having a `include` rule of `source.yaml` or `text.html`. why are you adding it?

**Review by octref — CHANGES_REQUESTED** [review]

File: packages/shiki/src/languages.ts (hunk 1 above)
Comment (LM1207) [reply packages/shiki/src/languages.ts]: You have already changed the file, I think I don't need to add the changes again.

**Comment by antfu** [comment]
Hey, thanks for your contribution! With the [recent rewrite of v1.0](https://github.com/shikijs/shiki/pull/557), we have moved the grammars out of this repository. We generally suggest you [loading the grammar locally](https://shiki.style/guide/load-lang) to move forward. If you believe this is the language that is widely used, you can send an issue or PR to this repo: https://github.com/shikijs/textmate-grammars-themes

Thank you!

**Pull request closed without merging**